cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java

```diff
@@ -178,16 +181,37 @@ private static void writeSubstations(Network network, String cimNamespace, XMLSt
             String finalGeoName = geoName;
             geographicalRegionIds.computeIfAbsent(geoName, k -> {
                 try {
-                    String subGeographicalRegionId = CgmesExportUtil.getUniqueId();
                     String geographicalRegionId = CgmesExportUtil.getUniqueId();
                     GeographicalRegionEq.write(geographicalRegionId, finalGeoName, cimNamespace, writer);
-                    SubGeographicalRegionEq.write(subGeographicalRegionId, finalGeoName, geographicalRegionId, cimNamespace, writer);
-                    return subGeographicalRegionId;
+                    return geographicalRegionId;
                 } catch (XMLStreamException e) {
                     throw new UncheckedXmlStreamException(e);
                 }
             });
-            SubstationEq.write(substation.getId(), substation.getNameOrId(), geographicalRegionIds.get(geoName), cimNamespace, writer);
+            String subGeographicalRegionId;
+            if (substation.hasProperty(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + "regionId")) {
```

Comment:
Similar proposal: separate in smaller methods that are responsible for getting the `subGeographicalRegionId` and the writing of the corresponding XML element. 
Something that could be read at this level as:
```java
subGeographicalRegionIds.computeIfAbsent(subGeographicalRegionIds(substation), 
    subGeoId -> writeGeographicalRegion(...));
```